include/smack/smack.h

```diff
@@ -57,49 +65,271 @@ int __SMACK_nondet() {
+  D("function {:inline} $sgt.i64(p1:i64, p2:i64) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i64(p1:i64, p2:i64) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i64(p1:i64, p2:i64) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $ult.i32(p1:i32, p2:i32) returns (bool) {p1 < p2}");
+  D("function {:inline} $ugt.i32(p1:i32, p2:i32) returns (bool) {p1 > p2}");
+  D("function {:inline} $ule.i32(p1:i32, p2:i32) returns (bool) {p1 <= p2}");
+  D("function {:inline} $uge.i32(p1:i32, p2:i32) returns (bool) {p1 >= p2}");
+  D("function {:inline} $slt.i32(p1:i32, p2:i32) returns (bool) {p1 < p2}");
+  D("function {:inline} $sgt.i32(p1:i32, p2:i32) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i32(p1:i32, p2:i32) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i32(p1:i32, p2:i32) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $ult.i16(p1:i16, p2:i16) returns (bool) {p1 < p2}");
+  D("function {:inline} $ugt.i16(p1:i16, p2:i16) returns (bool) {p1 > p2}");
+  D("function {:inline} $ule.i16(p1:i16, p2:i16) returns (bool) {p1 <= p2}");
+  D("function {:inline} $uge.i16(p1:i16, p2:i16) returns (bool) {p1 >= p2}");
+  D("function {:inline} $slt.i16(p1:i16, p2:i16) returns (bool) {p1 < p2}");
+  D("function {:inline} $sgt.i16(p1:i16, p2:i16) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i16(p1:i16, p2:i16) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i16(p1:i16, p2:i16) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $ult.i8(p1:i8, p2:i8) returns (bool) {p1 < p2}");
+  D("function {:inline} $ugt.i8(p1:i8, p2:i8) returns (bool) {p1 > p2}");
+  D("function {:inline} $ule.i8(p1:i8, p2:i8) returns (bool) {p1 <= p2}");
+  D("function {:inline} $uge.i8(p1:i8, p2:i8) returns (bool) {p1 >= p2}");
+  D("function {:inline} $slt.i8(p1:i8, p2:i8) returns (bool) {p1 < p2}");
+  D("function {:inline} $sgt.i8(p1:i8, p2:i8) returns (bool) {p1 > p2}");
+  D("function {:inline} $sle.i8(p1:i8, p2:i8) returns (bool) {p1 <= p2}");
+  D("function {:inline} $sge.i8(p1:i8, p2:i8) returns (bool) {p1 >= p2}");
+
+  D("function {:inline} $i2b(i: int) returns (bool) {i != 0}");
+  D("function {:inline} $b2i(b: bool) returns (i8) {if b then 1 else 0}");
+
+  D("function {:inline} $trunc.i64.i32(p: i64) returns (i32) {p}");
+  D("function {:inline} $trunc.i64.i16(p: i64) returns (i16) {p}");
+  D("function {:inline} $trunc.i64.i8(p: i64) returns (i8) {p}");
+  D("function {:inline} $trunc.i64.i1(p: i64) returns (int) {p}");
+  D("function {:inline} $trunc.i32.i16(p: i32) returns (i16) {p}");
+  D("function {:inline} $trunc.i32.i8(p: i32) returns (i8) {p}");
+  D("function {:inline} $trunc.i32.i1(p: i32) returns (int) {p}");
+  D("function {:inline} $trunc.i16.i8(p: i16) returns (i8) {p}");
+  D("function {:inline} $trunc.i16.i1(p: i16) returns (int) {p}");
+  D("function {:inline} $trunc.i8.i1(p: i8) returns (int) {p}");
+
+  D("function {:inline} $zext.i1.i64(p: int) returns (i64) {p}");
```

Comment:
And for many other declarations below. Note that the type `int` should never show up in the signature of a function that is models an LLVM operation, since LLVM does not have an `int` type, only `iXX`.